test_conformance/spirv_new/assemble_spirv.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+
+#####################################################################
+# Copyright (c) 2020 The Khronos Group Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#####################################################################
+
+"""Assembles the SPIR-V assembly files used by spirv_new into binaries,
+   and validates them using spirv-val.  Either run this from the parent
+   of the spirv_asm directory, or pass the --source-dir and --output-dir
+   options to specify the locations of the assembly files and the
+   binaries to be generated.
+"""
+
+import argparse
+import glob
+import os
+import sys
+from textwrap import wrap
+
+
+def fatal(message):
+    """Print an error message and exit with a non-zero status, to
+       indicate failure.
+    """
+    print(message)
+    sys.exit(1)
+
+
+def assemble_spirv(asm_dir, bin_dir, spirv_as, verbose):
+    """Assemble SPIR-V source into binaries."""
+
+    if not os.path.exists(bin_dir):
+        os.makedirs(bin_dir)
+
+    assembly_failures = False
+
+    for asm_file_path in glob.glob(os.path.join(asm_dir, '*.spvasm*')):
+        asm_file = os.path.basename(asm_file_path)
+        if os.path.isfile(asm_file_path):
+            if verbose:
+                print(' Assembling {}'.format(asm_file))
+
+            asm_file_root, asm_file_ext = os.path.splitext(asm_file)
+            bin_file = asm_file_root + asm_file_ext.replace('asm', '')
+            bin_file_path = os.path.join(bin_dir, bin_file)
+
+            command = '"{}" --target-env spv1.0 "{}" -o "{}"'.format(
+                spirv_as, asm_file_path, bin_file_path)
+            if os.system(command) != 0:
```

Comment:
Can we change executing command to call subprocess method `return_code = subprocess.call(command, shell=True) ` ?
This system method does not work on my windows 10. I got `The system cannot find the path specified.` error message. Probably some issues with escape characters in the path.
The same in validator part.

Reply:
Sounds reasonable. Will do that.

Reply:
Have done this. Does it look okay to you?

Reply:
@gwawiork Could you review the updated change please?

Reply:
Thanks for that it works. I noticed another issue. Looks like validator cannot find files in subdirectory where binary files have been saved
Example:
`python.exe assemble_spirv.py -s spirv_asm -o spirv_bin -a spirv-as.exe -l spirv-val.exe`

I have spirv-as and spirv-val in one directory. I have 2 subdirectories spirv_asm and spirv_bin. Assembling works ok but validation not. Please fix.
Example:

Validating fmod_float4.spv64
The system cannot find the path specified.
ERROR: Failure validating fmod_float4.spv64: see above output.

Reply:
I've pushed another change. Could you test it, please?